### ISSUE TITLE
New Rule: `no-literal-error-cause`

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16323]
### Rule details

Restricts to use literals as `cause` of `Error` like [`no-throw-literal`](https://eslint.org/docs/latest/rules/no-throw-literal).
[`Error.prototype.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) can contain any value, but for the purpose of the feature it is better to contain only Error. TypeScript doesn't solve this problem because the type of `Error.prototype.cause` is `unknown | undefined`.

### Related ECMAScript feature

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

### What type of rule is this?

Warns about a potential problem

### Example code

```js
try {
  connectToDatabase();
} catch (err) {
  throw new Error(
    'Connecting to database failed.',
    {
      // cause: err
      cause: err.message
    }
  );
}
```

### Why should this rule be in the core instead of a plugin?

`Error.prototype.cause` is a core feature of ECMAScript

### Participation

- [X] I am willing to submit a pull request to implement this rule.

**Comment by nzakas** [comment]
Thanks for the suggestion. This is actually more of a type-checking rule, which don't do for core rules. You can definitely create a custom rule if this is important to you.
